lib/src/instances/stream_monad.dart

```diff
@@ -68,10 +84,109 @@ class StreamMonad<T> extends Monad<T> implements Stream<T> {
   StreamMonad<E> asyncMap<E>(FutureOr convert(T event)) =>
       new StreamMonad(_stream.asyncMap(convert));
 
+  /// Combines multiple Observables to create an Observable whose values are
+  /// calculated from the latest values of each of its input Observables.
+  StreamMonad<E> combineLatest<S, E>(
+      StreamMonad<S> other, E combine(T element, S otherElement)) {
+    final combined = new StreamController<E>.broadcast();
+
+    T firstElement;
+    S otherFirstElement;
+    StreamSubscription<T> thisSubscription;
+    StreamSubscription<S> otherSubscription;
+
+    // ignore: type_annotate_public_apis, This is not public!
+    void onError(error, StackTrace stacktrace) {
+      if (combined.isClosed) {
+        return;
+      }
+      combined.addError(error, stacktrace);
+    }
+
+    void onDone() {
+      thisSubscription.cancel();
+      otherSubscription.cancel();
+      combined.close();
+    }
+
+    void processT(T element) {
+      firstElement = element;
+      if (combined.isClosed || otherFirstElement == null) {
+        return;
+      }
+      combined.add(combine(firstElement, otherFirstElement));
+    }
+
+    void processS(S element) {
+      otherFirstElement = element;
+      if (combined.isClosed || firstElement == null) {
+        return;
+      }
+      combined.add(combine(firstElement, otherFirstElement));
+    }
+
+    thisSubscription = listen(processT, onError: onError, onDone: onDone);
+    otherSubscription =
+        other.listen(processS, onError: onError, onDone: onDone);
+
+    return new StreamMonad(combined.stream);
+  }
+
   @override
   FutureMonad<bool> contains(Object needle) =>
       new FutureMonad(_stream.contains(needle));
 
+  /// Emits a value from the source Observable only after a particular time span
```

Comment:
the comment says the time span is determined by another observable, but here the duration is sent as an argument.